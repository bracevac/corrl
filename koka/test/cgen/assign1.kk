struct counter<h>( count : ref<h,int> )

function count( c : counter<h>, assigned : int) : write<h> () { 
  c.assign1/count.set(assigned) 
}

function main() {
  val c = Counter( count = ref(0) )
  c.count := 1
  c.count.set := 2
  print(!(c.count))
}
