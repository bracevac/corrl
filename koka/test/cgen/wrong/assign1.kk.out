test/cgen/wrong/assign1.kk(6, 5): error: no function count is defined that matches the argument types
 context : c.count := 1
 term : count
 inferred type: (counter<_h>, _a, ...) -> ...
 candidates : count: forall<h> (counter : counter<h>) -> ref<h,int>
 count: (s : string, pattern : string) -> int
 count: (s : string, pred : (char) -> bool) -> int
 count: (slice : sslice) -> int
 count: (s : string) -> int